src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -40,16 +60,107 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
         return false;
     }
 
-    public async loadMoreChildrenImpl(_clearCache: boolean): Promise<PostgresSchemaTreeItem[]> {
-        const user: string = nonNullProp(process.env, 'POSTGRES_USERNAME');
-        const fullUsername = user + "@" + nonNullProp(this.parent.server, 'name');
-        const password: string = nonNullProp(process.env, 'POSTGRES_PASSWORD');
-        const sslString = process.env.POSTGRES_SSL;
-        const ssl: boolean = sslString === 'true';
-        const host: string = nonNullProp(this.parent.server, 'fullyQualifiedDomainName');
-        const clientConfig: ClientConfig = { user: fullUsername, password: password, ssl: ssl, host: host, port: 5432, database: this.databaseName };
-        const accountConnection = new Client(clientConfig);
-        const db = await pgStructure(accountConnection);
-        return db.schemas.map(schema => new PostgresSchemaTreeItem(this, schema));
+    public async loadMoreChildrenImpl(_clearCache: boolean): Promise<AzExtTreeItem[]> {
+        try {
+            const { username, password } = await this.getCredentials(false, true);
+
+            const sslString: string | undefined = process.env.POSTGRES_SSL;
+            const ssl: boolean = sslString === 'true';
+            const host: string = nonNullProp(this.parent.server, 'fullyQualifiedDomainName');
+            const clientConfig: ClientConfig = { user: username, password, ssl, host, port: 5432, database: this.databaseName };
+            const accountConnection: Client = new Client(clientConfig);
+            const db: Db = await pgStructure(accountConnection);
+            return db.schemas.map(schema => new PostgresSchemaTreeItem(this, schema));
+        } catch (error) {
+            const parsedError: IParsedError = parseError(error);
+
+            if (parsedError.errorType !== 'UserCancelledError') {
+                // tslint:disable-next-line: no-floating-promises
+                ext.ui.showWarningMessage(localize('couldNotConnect', 'Could not connect to "{0}": {1}', this.parent.label, parsedError.message));
+            }
+
+            return [new GenericTreeItem(this, {
+                contextValue: 'postgresCredentials',
+                label: localize('enterCredentials', 'Enter server credentials to connect to "{0}"...', this.parent.label),
+                commandId: 'cosmosDB.getPostgresCredentials'
+            })];
+        }
+    }
+
+    public async getCredentials(forcePrompt: boolean, warnBeforePrompting: boolean): Promise<{ username: string, password: string }> {
+        let username: string | undefined;
+        let password: string | undefined;
+
+        if (!forcePrompt) {
+            const storedValue: string | undefined = ext.context.globalState.get(this._serviceName);
```

Comment:
I think this logic is complicated enough to warrant it's own function. Maybe like `getCredentialsFromKeytar` or something